frontends/common/constantFolding.h

```diff
@@ -23,36 +23,97 @@ limitations under the License.
 #include "frontends/p4/typeChecking/typeChecker.h"
 
 namespace P4 {
-// Can also be run before type checking, so it only uses types if
-// they are available.
+
+/** Implements a pass that statically evaluates many constant
+ * expressions.
+ *
+ * Currently the following expressions are treated as constants:
+ *  - `IR::Constant`
+ *  - `IR::BoolLiteral`
+ *  - `IR::ListExpression` where each element is also a constant
+ *  - `IR::TypeNameExpression`
+ *
+ * The pass can be invoked either with or without the reference and
+ * type maps. When this information is not available, then several
+ * cases revert to the identify function.
+ *
+ * The pass also eliminates arbitrary-precision integer literals,
+ * replacing them with literals that have explicit widths.
```

Comment:
Mihai: This is incorrect, that is done by the type checker. There isn't enough information to do this here.